sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ExchangeStatisticsCollector.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.exchange
+
+sealed trait ExchangeStatisticsSpec
+
+case object NullPartitionKeyStatisticsSpec extends ExchangeStatisticsSpec
+
+trait ExchangeStatisticsCollector {
+
+  /**
+   * types of Spec to indicate which ExchangeStatistics need to be collected.
+   */
+  def statSpecs: Seq[ExchangeStatisticsSpec]
+
+  /**
+   * return whether an Exchange knows how many NullPartitionKey records.
```

Comment:
`NullPartitionKey`?

Reply:
The naming is tricky?
basically i want to present the NullPartitionKey to the following logical, while in NAAJ case, I want to count the records which is all null in project UnsafeRow key, and NAAJ support singleColumn only, the following code only check on the first column of HashPartitioning.expressions.head

But I've had no better option, maybe you can offer a better one?

```
val projection = UnsafeProjection.create(
          h.partitionIdExpression :: h.expressions.head :: Nil, outputAttributes)
        row => {
          val projectedRow = projection(row)
          // Update NULL_PARTITION_KEY_RECORDS_WRITTEN metrics if nullPartitionKey show up.
          if (projectedRow.isNullAt(1)) {
            writeMetrics(NULL_PARTITION_KEY_RECORDS_WRITTEN).add(1L)
          }
```

Reply:
I meant `NullPartitionKey` is not a class or object or similar thing you introduced, I think you mean "how many records with all null values of partition keys"? I think we should make the doc clear to readers.

When reading this doc, as a reader I'd wonder what `NullPartitionKey` is but cannot find related class/object.